Library/Homebrew/shims/super/cc

```diff
@@ -310,6 +311,9 @@ class Cmd
     args += path_flags("-isystem", isystem_paths) + path_flags("-I", include_paths)
     # Add -nostdinc when building against glibc@2.13 to avoid mixing system and brewed glibc headers.
     args << "-nostdinc" if @deps.include?("glibc@2.13")
+    # Ideally this would be -ffile-prefix-map, but that requires a minimum of GCC 8, LLVM Clang 10 or Apple Clang 12
+    # and detecting the version dynamcally based on what `HOMEBREW_CC` may have been rewritten to point to is awkward
```

Comment:
We could just take a page from a `configure` script and see if passing the flag doesn't error...

Reply:
Probably don't want to do that on every `cc` run and doing it once is tricky because `HOMEBREW_CC` can and does change during the built (e.g. swift wrapper).

I guess ideally what we'd do is query all compiler versions ahead of time and pick the appropriate one. Not really sure...

Reply:
Hmmm, yea. What cases will doing this for macOS with Xcode >= 12 going to cause problems?

Reply:
If we check CLT too then it's probably OK, though it's a bit messier on the GCC side (which includes Linux) where we technically still support GCC 4.9 and later in brew.